src/main/java/harper/utils/Storage.java

```diff
@@ -0,0 +1,107 @@
+    private static final String PROJECT_DIR = System.getProperty("user.dir");
+    private File file;
+
+    //CHECKSTYLE.OFF: MissingJavadocMethod
+    public Storage(String folderName, String fileName) {
+        File folder = new File(PROJECT_DIR, folderName);
+        try {
+            if (!folder.exists()) {
+                folder.mkdir();
+            }
+            File file = new File(folder, fileName);
+            this.file = file;
+            if (!file.exists()) {
+                file.createNewFile();
+                this.file = file;
+            }
+        } catch (IOException | SecurityException e) {
+            throw new HarperFileCreatingException();
+        }
+    }
+
+    /**
+     * Loads data from hard disk.
+     *
+     * @return An ArrayList contains the data from hard disk.
+     */
+    public ArrayList<Task> load() {
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            Scanner scanner = new Scanner(this.file);
+            while (scanner.hasNextLine()) {
+                String[] taskString = scanner.nextLine().split(" \\| ");
+                boolean isDone;
+                if (taskString[1].equals("1")) {
+                    isDone = true;
+                } else if (taskString[1].equals("0")) {
+                    isDone = false;
+                } else {
+                    throw new HarperFileLoadingException();
+                }
+
+                if (taskString[0].equals("T")) {
+                    Task task = new ToDo(taskString[2], isDone);
+                    tasks.add(task);
+                } else if (taskString[0].equals("D")) {
+                    String by = taskString[3];
+                    LocalDateTime byFormatted = LocalDateTime.parse(by, DATE_TIME_FORMATTER);
+                    Task task = new Deadline(taskString[2], isDone, byFormatted);
+                    tasks.add(task);
+                } else if (taskString[0].equals("E")) {
+                    String[] duration = taskString[3].split(" - ");
+                    LocalDateTime start = LocalDateTime.parse(duration[0], DATE_TIME_FORMATTER);
+                    LocalDateTime end = LocalDateTime.parse(duration[1], DATE_TIME_FORMATTER);
+                    Task task = new Event(taskString[2], isDone, start, end);
+                    tasks.add(task);
+                } else {
+                    throw new HarperFileLoadingException();
+                }
+            }
+            scanner.close();
+            return tasks;
+        } catch (FileNotFoundException | ArrayIndexOutOfBoundsException | DateTimeParseException e) {
+            throw new HarperFileLoadingException();
+        }
+    }
+
+    /**
+     * Stores the task list into hard disk.
+     *
+     * @param taskList task.Task list to be stored.
+     */
+    public void save(TaskList taskList) {
+        try {
+            FileWriter fw = new FileWriter(this.file);
```

Comment:
Should this be named 'fileWriter' instead? Makes it easier to pronounce as mentioned [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/topics.html#w4-6-code-quality-naming)